From: Tobias Quathamer <toddy@debian.org>
Date: Fri, 24 Feb 2012 14:29:11 +0100
Subject: Fix typos spotted by Lintian

---
 atc/update.c |    2 +-
 hack/hack.c  |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/atc/update.c b/atc/update.c
index 3a145e5..39bbf2a 100644
--- a/atc/update.c
+++ b/atc/update.c
@@ -153,7 +153,7 @@ update(dummy)
 		}
 		if (pp->altitude > 9)
 			/* "this is impossible" */
-			loser(pp, "exceded flight ceiling.");
+			loser(pp, "exceeded flight ceiling.");
 		if (pp->altitude <= 0) {
 			for (i = 0; i < sp->num_airports; i++)
 				if (pp->xpos == sp->airport[i].x &&
diff --git a/hack/hack.c b/hack/hack.c
index 05141af..6e5028e 100644
--- a/hack/hack.c
+++ b/hack/hack.c
@@ -527,7 +527,7 @@ pickup(int all)
 			}
 	lift_some:
 			if (inv_cnt() >= 52) {
-				pline("Your knapsack cannot accomodate anymore items.");
+				pline("Your knapsack cannot accommodate anymore items.");
 				break;
 			}
 			if (wt > -5)
-- 
